net/src/main/java/org/apollo/net/codec/update/OnDemandResponseEncoder.java

```diff
@@ -0,0 +1,50 @@
+package org.apollo.net.codec.update;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+import io.netty.handler.codec.MessageToMessageEncoder;
+
+/**
+ * A {@link MessageToMessageEncoder} for the 'on-demand' protocol.
+ *
+ * @author Graham
+ */
+public final class OnDemandResponseEncoder extends MessageToByteEncoder<OnDemandResponse> {
+
+	@Override
+	protected void encode(ChannelHandlerContext ctx, OnDemandResponse response, ByteBuf buf) {
+		int fs = response.getFs();
+		int folder = response.getFolder();
+		ByteBuf container = response.getChunkData();
+
+		buf.writeByte(fs);
+		buf.writeShort(folder);
+
+		int compression = container.readUnsignedByte();
+		if (response.getPriority() == OnDemandRequest.Priority.LOW) {
+			compression |= 0x80;
```

Comment:
These magic numbers need to go (including the 508, 511 below).

Reply:
I've added constants that are representative descriptions for what they're used for.